test/cmark-fuzz.c

```diff
@@ -1,28 +1,76 @@
 #include <stdint.h>
 #include <stdlib.h>
+#include <string.h>
 #include "cmark-gfm.h"
+#include "cmark-gfm-core-extensions.h"
+
+const char *extension_names[] = {
+  "autolink",
+  "strikethrough",
+  "table",
+  "tagfilter",
+  NULL,
+};
+
+int LLVMFuzzerInitialize(int *argc, char ***argv) {
+  cmark_gfm_core_extensions_ensure_registered();
+  return 0;
+}
 
 int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
-  int options = 0;
-  if (size > sizeof(options)) {
-    /* First 4 bytes of input are treated as options */
-    int options = *(const int *)data;
+  struct __attribute__((packed)) {
+    int options;
+    int width;
+  } fuzz_config;
+
+  if (size >= sizeof(fuzz_config)) {
+    /* The beginning of `data` is treated as fuzzer configuration */
+    memcpy(&fuzz_config, data, sizeof(fuzz_config));
 
     /* Mask off valid option bits */
-    options = options & (CMARK_OPT_SOURCEPOS | CMARK_OPT_HARDBREAKS | CMARK_OPT_UNSAFE | CMARK_OPT_NOBREAKS | CMARK_OPT_NORMALIZE | CMARK_OPT_VALIDATE_UTF8 | CMARK_OPT_SMART);
+    fuzz_config.options &= (
+      CMARK_OPT_SOURCEPOS |
+      CMARK_OPT_HARDBREAKS |
+      CMARK_OPT_NOBREAKS |
+      CMARK_OPT_NORMALIZE |
+      CMARK_OPT_VALIDATE_UTF8 |
+      CMARK_OPT_SMART |
+      /* GFM specific options */
+      CMARK_OPT_GITHUB_PRE_LANG |
+      CMARK_OPT_LIBERAL_HTML_TAG |
+      CMARK_OPT_FOOTNOTES |
+      CMARK_OPT_STRIKETHROUGH_DOUBLE_TILDE |
+      CMARK_OPT_TABLE_PREFER_STYLE_ATTRIBUTES |
+      CMARK_OPT_FULL_INFO_STRING |
+      CMARK_OPT_UNSAFE
```

Comment:
Is there anything we can / should do (besides developer awareness) to keep this list up to date when new extensions / config are added? Is there any documentation or CI that we could add?

Reply:
It would be super ideal if we could. I could add a CI check for it? grep for all used `CMARK_OPT_` defines in the codebase, check all are mentioned in this file. What do you think?

Reply:
These are defined in `src/cmark-gfm.h` so the usual approach would be to define a mask of all valid options there:
```diff
  * replaced by a placeholder HTML comment, and unsafe links are replaced by
  * empty strings.
  */
 #define CMARK_OPT_UNSAFE (1 << 17)

+#define CMARK_OPT_MASK ( \
+    CMARK_OPT_SOURCEPOS | \
+    CMARK_OPT_HARDBREAKS | \
+    CMARK_OPT_NOBREAKS | \
+    etc
```

I originally masked off the options because passing invalid options to libraries can sometimes cause weird issues. However, I just looked at the implementation of  `cmark_parser_new` and it doesn't validate any of the options anyway. So, I think the easiest solution is to just kill the whole masking logic